napari/utils/events/evented_model.py

```diff
@@ -91,6 +97,26 @@ def __new__(mcs, name, bases, namespace, **kwargs):
                 # required for pydantic>=1.8.0 due to:
                 # https://github.com/samuelcolvin/pydantic/pull/2064
                 EventedModel.__config__.json_encoders[f.type_] = encoder
+        # check for @_.setters defined on the class, so we can allow them
+        # in EventedModel.__setattr__
+        cls.__property_setters__ = {}
+        for name, attr in namespace.items():
+            if isinstance(attr, property) and attr.fset is not None:
+                cls.__property_setters__[name] = attr
+
+        cls.__field_dependents__ = {}
+        for prop, fields in _deps.items():
+            if prop not in cls.__property_setters__:
+                raise ValueError(
+                    'Fields with dependencies must be property.setters. '
+                    f'{prop!r} is not.'
+                )
+            for field in fields:
+                if field in cls.__field_dependents__:
```

Comment:
Should we check that `field` is strictly a Pydantic field? Or do we want to allow dependencies between mutable properties? I.e. do we want to allow something like this:

```python
    class T(EventedModel, dependencies={'b': ['a'], 'c': ['b']}):
        a: int = 1
        @property
        def b(self) -> int:
            return self.a + 1
        @b.setter
        def b(self, b: int):
            self.a = b - 1
        @property
        def c(self) -> int:
            return self.b + 1
        @c.setter
        def c(self, c: int):
            self.b = c - 1
```

Reply:
i like this, i actually added this locally (and moved the `dependencies` kwarg to the `class Config` defined on the model)

Reply:
just pushed a change that autodetects dependencies.  So adding it to config is now an optional override